### PR TITLE
Modify spark image to include azure and gcs dependency jars

### PR DESCRIPTION
**Pull request by anusudarsan**
Modify the spark testing image to include Azure and Google cloud storage hadoop dependency jars to enable querying their object stores

### CODE REVIEW

**Comment by anusudarsan** [comment]
any ideas on the failures? I was able to build locally by adding a trusted host flag for pypi, not sure if this is what we want for CI or we have an expired ssl cert

**Comment by nineinchnick** [comment]
Looks like it started failing on the builds 2 weeks ago: https://github.com/trinodb/docker-images/actions/runs/9024908079

It's probably caused by using CentOS 7 as the base image, which might not getting any updates anymore. We should update `pip`, but I'm not yet sure what's the best way to do that.

**Comment by anusudarsan** [comment]
I added a new commit to https://github.com/trinodb/trino/pull/21958 for testing gcs/azure for spark compatibility

**Comment by ebyhr** [comment]
Just rebased on master to run with https://github.com/trinodb/docker-images/pull/199

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Started release job: https://github.com/trinodb/docker-images/actions/runs/9220333116

File: testing/spark3-iceberg/Dockerfile [hunk 1]
```diff
@@ -35,6 +35,12 @@ WORKDIR ${SPARK_HOME}/jars
 # install AWS SDK so we can access S3; the version must match the hadoop-* jars which are part of SPARK distribution
 RUN wget -nv "https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/3.3.4/hadoop-aws-3.3.4.jar"
 RUN wget -nv "https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-bundle/1.12.319/aws-java-sdk-bundle-1.12.319.jar"
+# install Azure SDK so we can access azure file system; the version must match the hadoop-* jars which are part of SPARK distribution
+RUN wget -nv https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-azure/3.3.4/hadoop-azure-3.3.4.jar
+RUN wget -nv https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-azure-datalake/3.3.6/hadoop-azure-datalake-3.3.6.jar
+RUN wget -nv https://repo1.maven.org/maven2/com/microsoft/azure/azure-storage/8.6.6/azure-storage-8.6.6.jar
+# install Google Hadoop connector so we can access gcs
+RUN wget -nv https://storage.googleapis.com/hadoop-lib/gcs/gcs-connector-hadoop2-latest.jar
```
Comment (findinpath) [inline testing/spark3-iceberg/Dockerfile:43]: Can't we use a pinned version of this dependency?

**Pull request merged**